I think there is something missing in the patch. I think that you probably need to replace `ResolvableProfile` by  `SeparateResolvableProfile` in `build-scripts/compile_profiles.py:43`. Currently I'm getting a traceback during the build if I try to use the controls in a profile in a real product:

```
Traceback (most recent call last):                                                                                                                                                            
  File "/home/jcerny/work/git/scap-security-guide/ssg/build_yaml.py", line 640, in add_profiles_from_dir                                                                                          new_profile = Profile.from_yaml(dir_item_path, env_yaml)                                                                                                                                  
  File "/home/jcerny/work/git/scap-security-guide/ssg/build_yaml.py", line 139, in from_yaml                                                                                                  
    raise RuntimeError("Unparsed YAML data in '%s'.\n\n%s"                                                                                                                                    RuntimeError: Unparsed YAML data in '/home/jcerny/work/git/scap-security-guide/rhel8/./profiles/abcd-low.profile'.                                                                            
                                                                                                                                                                                              
{'policies': [{'id': 'abcd', 'controls': ['R1', 'R2', 'R3']}]}                                                                                                                                                                                                                                                                                                                              
During handling of the above exception, another exception occurred:                                                                                                                           
                                                                                                                                                                                              Traceback (most recent call last):                                                                                                                                                            
  File "/home/jcerny/work/git/scap-security-guide/build-scripts/yaml_to_shorthand.py", line 95, in <module>                                                                                   
    main()                                                                                                                                                                                      File "/home/jcerny/work/git/scap-security-guide/build-scripts/yaml_to_shorthand.py", line 89, in main                                                                                       
    loader.process_directory_tree(benchmark_root, add_content_dirs)                                                                                                                           
  File "/home/jcerny/work/git/scap-security-guide/ssg/build_yaml.py", line 1335, in process_directory_tree                                                                                        self._load_group_process_and_recurse(start_dir)                                                                                                                                           
  File "/home/jcerny/work/git/scap-security-guide/ssg/build_yaml.py", line 1321, in _load_group_process_and_recurse                                                                           
    self.loaded_group = self.load_benchmark_or_group(guide_directory)                                                                                                                           File "/home/jcerny/work/git/scap-security-guide/ssg/build_yaml.py", line 1419, in load_benchmark_or_group                                                                                   
    result = super(ListInputsLoader, self).load_benchmark_or_group(guide_directory)                                                                                                           
  File "/home/jcerny/work/git/scap-security-guide/ssg/build_yaml.py", line 1311, in load_benchmark_or_group                                                                                       group.add_profiles_from_dir(self.action, self.profiles_dir, self.env_yaml)                                                                                                                
  File "/home/jcerny/work/git/scap-security-guide/ssg/build_yaml.py", line 646, in add_profiles_from_dir                                                                                      
    raise RuntimeError(msg)                                                                                                                                                                   RuntimeError: Error building profile from '/home/jcerny/work/git/scap-security-guide/rhel8/./profiles/abcd-low.profile': 'Unparsed YAML data in '/home/jcerny/work/git/scap-security-guide/rhe
l8/./profiles/abcd-low.profile'.                                                                                                                                                              
                                                                                                                                                                                              {'policies': [{'id': 'abcd', 'controls': ['R1', 'R2', 'R3']}]}'  
```

--- a/ssg/build_yaml.py
+++ b/ssg/build_yaml.py
@@ -332,18 +332,15 @@ class ResolvableProfile(Profile):
         self.resolved = False
         self.resolved_selections = set()
 
-    def _controls_to_items(self, controls_manager, policy_id, control_id_list):
-        items = Control()
-        for control_id in control_id_list:
-            new_control = controls_manager.get_control(policy_id, control_id)
-            items.rules.extend(new_control.rules)
-            items.variables.update(new_control.variables)
-        items.rules = list(set(items.rules))
+    def _controls_ids_to_controls(self, controls_manager, policy_id, control_id_list):
+        items = [controls_manager.get_control(policy_id, cid) for cid in control_id_list]
         return items
 
-    def _policy_to_items(self, controls_manager, policy_id):
-        control_id_list = [c.id for c in controls_manager.get_all_controls(policy_id)]
-        return self._controls_to_items(controls_manager, policy_id, control_id_list)
+    def _merge_control(self, control):
+        self.selected.extend(control.rules)
+        for varname, value in control.variables.items():
+            if varname not in self.variables:
+                self.variables[varname] = value
 
     def resolve_controls(self, controls_manager):
         pass
@@ -406,31 +403,29 @@ class ProfileWithSeparatePolicies(ResolvableProfile):
     def _parse_policies(self, policies_yaml):
         for item in policies_yaml:
             id_ = required_key(item, "id")
-            controls = required_key(item, "controls")
-            if not isinstance(controls, list):
-                if controls != "all":
+            controls_ids = required_key(item, "controls")
+            if not isinstance(controls_ids, list):
+                if controls_ids != "all":
                     msg = ("Policy {id_} contains invalid controls list {controls}."
-                        .format(id_=id_, controls=str(controls))
+                        .format(id_=id_, controls=str(controls_ids))
                         )
                     raise ValueError(msg)
-            self.policies[id_] = controls
+            self.policies[id_] = controls_ids
 
     def resolve_controls(self, controls_manager):
-        for policy_id, controls in self.policies.items():
-            to_expand = Control()
+        for policy_id, controls_ids in self.policies.items():
+            controls = []
 
-            if isinstance(controls, list):
-                to_expand = self._controls_to_items(controls_manager, policy_id, controls)
-            elif controls == "all":
-                to_expand = self._policy_to_items(controls_manager, policy_id)
+            if isinstance(controls_ids, list):
+                controls = self._controls_ids_to_controls(controls_manager, policy_id, controls_ids)
+            elif controls_ids == "all":
+                controls = controls_manager.get_all_controls(policy_id)
             else:
-                msg = "Unknown policy content {content} in profile {profile_id}".format(content=controls, profile_id=self.id_)
+                msg = "Unknown policy content {content} in profile {profile_id}".format(content=controls_ids, profile_id=self.id_)
                 raise ValueError(msg)
 
-            self.selected.extend(to_expand.rules)
-            for varname, value in to_expand.variables.items():
-                if varname not in self.variables:
-                    self.variables[varname] = value
+            for c in controls:
+                self._merge_control(c)
 
     def extend_by(self, extended_profile):
         self.policies.update(extended_profile.policies)
@@ -450,16 +445,16 @@ class ProfileWithInlinePolicies(ResolvableProfile):
             super(ProfileWithInlinePolicies, self).apply_selection(item)
 
     def resolve_controls(self, controls_manager):
-        for policy_id, controls in self.controls_by_policy.items():
-            if "all" in controls:
-                to_expand = self._policy_to_items(controls_manager, policy_id)
-            else:
-                to_expand = self._controls_to_items(controls_manager, policy_id, controls)
+        for policy_id, controls_ids in self.controls_by_policy.items():
+            controls = []
 
-            self.selected.extend(to_expand.rules)
-            for varname, value in to_expand.variables.items():
-                if varname not in self.variables:
-                    self.variables[varname] = value
+            if "all" in controls_ids:
+                controls = controls_manager.get_all_controls(policy_id)
+            else:
+                controls = self._controls_ids_to_controls(controls_manager, policy_id, controls_ids)
+
+            for c in controls:
+                self._merge_control(c)
 
 
 class Value(object):

--- a/ssg/build_yaml.py
+++ b/ssg/build_yaml.py
@@ -326,96 +326,23 @@ class Profile(object):
         return profile
 
 
-class ProfileWithSeparatePolicies(Profile):
-    def __init__(self, * args, ** kwargs):
-        super(ProfileWithSeparatePolicies, self).__init__(* args, ** kwargs)
-        self.policies = {}
-
-    def read_yaml_contents(self, yaml_contents):
-        policies = yaml_contents.pop("policies", None)
-        if policies:
-            self._parse_policies(policies)
-        super(ProfileWithSeparatePolicies, self).read_yaml_contents(yaml_contents)
-
-    def _parse_policies(self, policies_yaml):
-        for item in policies_yaml:
-            id_ = required_key(item, "id")
-            controls = required_key(item, "controls")
-            if not isinstance(controls, list):
-                if controls != "all":
-                    msg = ("Policy {id_} contains invalid controls list {controls}."
-                        .format(id_=id_, controls=str(controls))
-                        )
-                    raise ValueError(msg)
-            self.policies[id_] = controls
-
-    def resolve_controls(self, controls_manager):
-        for policy_id, controls in self.policies.items():
-            to_expand = Control()
-
-            if isinstance(controls, list):
-                to_expand = self._controls_to_items(controls_manager, policy_id, controls)
-            elif controls == "all":
-                to_expand = self._policy_to_items(controls_manager, policy_id)
-            else:
-                msg = "Unknown policy content {content} in profile {profile_id}".format(content=controls, profile_id=self.id_)
-                raise ValueError(msg)
-
-            self.selected.extend(to_expand.rules)
-            for varname, value in to_expand.variables.items():
-                if varname not in self.variables:
-                    self.variables[varname] = value
-
-    def extend_by(self, extended_profile):
-        self.policies.update(extended_profile.policies)
-        super(ProfileWithSeparatePolicies, self).extend_by(extended_profile)
-
-
-class ProfileWithInlinePolicies(Profile):
-    def __init__(self, * args, ** kwargs):
-        super(ProfileWithInlinePolicies, self).__init__(* args, ** kwargs)
-        self.controls_by_policy = defaultdict(list)
-
-    def apply_selection(self, item):
-        if ":" in item:
-            policy_id, control_id = item.split(":", 1)
-            self.controls_by_policy[policy_id].append(control_id)
-        else:
-            super(ProfileWithInlinePolicies, self).apply_selection(item)
-
-    def resolve_controls(self, controls_manager):
-        for policy_id, controls in self.controls_by_policy.items():
-            if "all" in controls:
-                to_expand = self._policy_to_items(controls_manager, policy_id)
-            else:
-                to_expand = self._controls_to_items(controls_manager, policy_id, controls)
-
-            self.selected.extend(to_expand.rules)
-            for varname, value in to_expand.variables.items():
-                if varname not in self.variables:
-                    self.variables[varname] = value
-
-
 class ResolvableProfile(Profile):
     def __init__(self, * args, ** kwargs):
         super(ResolvableProfile, self).__init__(* args, ** kwargs)
         self.resolved = False
         self.resolved_selections = set()
 
-    def _control_to_items(self, controls_manager, policy_id, control_id):
-        return controls_manager.get_control(policy_id, control_id)
-
     def _controls_to_items(self, controls_manager, policy_id, control_id_list):
         items = Control()
         for control_id in control_id_list:
-            new_control = self._control_to_items(controls_manager, policy_id, control_id)
+            new_control = controls_manager.get_control(policy_id, control_id)
             items.rules.extend(new_control.rules)
             items.variables.update(new_control.variables)
         items.rules = list(set(items.rules))
         return items
 
     def _policy_to_items(self, controls_manager, policy_id):
-        control_id_list = controls_manager.get_all_controls(policy_id)
+        control_id_list = [c.id for c in controls_manager.get_all_controls(policy_id)]
         return self._controls_to_items(controls_manager, policy_id, control_id_list)
 
     def resolve_controls(self, controls_manager):
@@ -463,6 +390,76 @@ class ResolvableProfile(Profile):
         self.selected = sorted(self.resolved_selections)
 
         self.resolved = True
+
+
+class ProfileWithSeparatePolicies(ResolvableProfile):
+    def __init__(self, * args, ** kwargs):
+        super(ProfileWithSeparatePolicies, self).__init__(* args, ** kwargs)
+        self.policies = {}
+
+    def read_yaml_contents(self, yaml_contents):
+        policies = yaml_contents.pop("policies", None)
+        if policies:
+            self._parse_policies(policies)
+        super(ProfileWithSeparatePolicies, self).read_yaml_contents(yaml_contents)
+
+    def _parse_policies(self, policies_yaml):
+        for item in policies_yaml:
+            id_ = required_key(item, "id")
+            controls = required_key(item, "controls")
+            if not isinstance(controls, list):
+                if controls != "all":
+                    msg = ("Policy {id_} contains invalid controls list {controls}."
+                        .format(id_=id_, controls=str(controls))
+                        )
+                    raise ValueError(msg)
+            self.policies[id_] = controls
+
+    def resolve_controls(self, controls_manager):
+        for policy_id, controls in self.policies.items():
+            to_expand = Control()
+
+            if isinstance(controls, list):
+                to_expand = self._controls_to_items(controls_manager, policy_id, controls)
+            elif controls == "all":
+                to_expand = self._policy_to_items(controls_manager, policy_id)
+            else:
+                msg = "Unknown policy content {content} in profile {profile_id}".format(content=controls, profile_id=self.id_)
+                raise ValueError(msg)
+
+            self.selected.extend(to_expand.rules)
+            for varname, value in to_expand.variables.items():
+                if varname not in self.variables:
+                    self.variables[varname] = value
+
+    def extend_by(self, extended_profile):
+        self.policies.update(extended_profile.policies)
+        super(ProfileWithSeparatePolicies, self).extend_by(extended_profile)
+
+
+class ProfileWithInlinePolicies(ResolvableProfile):
+    def __init__(self, * args, ** kwargs):
+        super(ProfileWithInlinePolicies, self).__init__(* args, ** kwargs)
+        self.controls_by_policy = defaultdict(list)
+
+    def apply_selection(self, item):
+        if ":" in item:
+            policy_id, control_id = item.split(":", 1)
+            self.controls_by_policy[policy_id].append(control_id)
+        else:
+            super(ProfileWithInlinePolicies, self).apply_selection(item)
+
+    def resolve_controls(self, controls_manager):
+        for policy_id, controls in self.controls_by_policy.items():
+            if "all" in controls:
+                to_expand = self._policy_to_items(controls_manager, policy_id)
+            else:
+                to_expand = self._controls_to_items(controls_manager, policy_id, controls)
+
+            self.selected.extend(to_expand.rules)
+            for varname, value in to_expand.variables.items():
+                if varname not in self.variables:
+                    self.variables[varname] = value
 
 
 class Value(object):

--- a/tests/unit/ssg-module/data/profiles_dir/abcd-all-inline.profile
+++ b/tests/unit/ssg-module/data/profiles_dir/abcd-all-inline.profile
@@ -1,0 +1,8 @@
+documentation_complete: true
+title: ABCD ALL for Red Hat Enterprise Linux 9
+description: >-
+  This profile contains configuration checks that align to
+  the ABCD benchmark.
+selections:
+  - abcd:all
+  - security_patches_uptodate

--- a/tests/unit/ssg-module/test_controls.py
+++ b/tests/unit/ssg-module/test_controls.py
@@ -62,28 +62,28 @@ def test_controls_load_product():
     assert "cockpit_session_timeout" in c_r1.rules
 
 
-class SeparateResolvableProfile(ssg.build_yaml.ProfileWithSeparatePolicies, ssg.build_yaml.ResolvableProfile):
-    pass
-
-
-class InlineResolvableProfile(ssg.build_yaml.ProfileWithInlinePolicies, ssg.build_yaml.ResolvableProfile):
-    pass
-
-
 def test_profile_resolution_separate():
-    profile_resolution(SeparateResolvableProfile, "abcd-low")
+    profile_resolution(ssg.build_yaml.ProfileWithSeparatePolicies, "abcd-low")
 
 
 def test_profile_resolution_extends_separate():
-    profile_resolution_extends(SeparateResolvableProfile, "abcd-low", "abcd-high")
+    profile_resolution_extends(ssg.build_yaml.ProfileWithSeparatePolicies, "abcd-low", "abcd-high")
+
+
+def test_profile_resolution_all_separate():
+    profile_resolution_all(ssg.build_yaml.ProfileWithSeparatePolicies, "abcd-all")
 
 
 def test_profile_resolution_inline():
-    profile_resolution(InlineResolvableProfile, "abcd-low-inline")
+    profile_resolution(ssg.build_yaml.ProfileWithInlinePolicies, "abcd-low-inline")
 
 
 def test_profile_resolution_extends_inline():
-    profile_resolution_extends(InlineResolvableProfile, "abcd-low-inline", "abcd-high-inline")
+    profile_resolution_extends(ssg.build_yaml.ProfileWithInlinePolicies, "abcd-low-inline", "abcd-high-inline")
+
+
+def test_profile_resolution_all_inline():
+    profile_resolution_all(ssg.build_yaml.ProfileWithInlinePolicies, "abcd-all-inline")
 
 
 def profile_resolution(cls, profile_low):
@@ -140,14 +140,13 @@ def profile_resolution_extends(cls, profile_low, profile_high):
     assert high_profile.variables["var_password_pam_ocredit"] == "2"
 
 
-@pytest.mark.skip
-def test_profile_resolution_all():
+def profile_resolution_all(cls, profile_all):
     env_yaml = {"product": "rhel9"}
     controls_manager = ssg.controls.ControlsManager(controls_dir, env_yaml)
     controls_manager.load()
-    profile_path = os.path.join(profiles_dir, "abcd-all.profile")
-    profile = ssg.build_yaml.ResolvableProfile.from_yaml(profile_path)
-    all_profiles = {"abcd-all": profile}
+    profile_path = os.path.join(profiles_dir, profile_all + ".profile")
+    profile = cls.from_yaml(profile_path)
+    all_profiles = {profile_all: profile}
     profile.resolve(all_profiles, controls_manager=controls_manager)
 
     # Profile 'abcd-all' selects all controls from 'abcd' policy,

